tests/bluetooth/tester/src/btp_mcp.c

```diff
@@ -0,0 +1,1307 @@
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t mcp_current_track_obj_id_set(const void *cmd, uint16_t cmd_len, void *rsp,
+					    uint16_t *rsp_len)
+{
+	const struct btp_mcp_current_track_obj_id_set_cmd *cp = cmd;
+	uint64_t id = sys_le64_to_cpu(cp->id);
+	struct bt_conn *conn;
+	int err;
+
+	LOG_DBG("MCC Set Current Track Object ID");
+
+	conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, &cp->address);
+	if (!conn) {
+		LOG_ERR("Unknown connection");
+		return BTP_STATUS_FAILED;
+	}
+
+	err = bt_mcc_set_current_track_obj_id(conn, id);
+	if (err) {
+		return BTP_STATUS_FAILED;
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t mcp_cmd_send(const void *cmd, uint16_t cmd_len, void *rsp, uint16_t *rsp_len)
+{
+	const struct btp_mcp_send_cmd *cp = cmd;
+	struct mpl_cmd mcp_cmd;
+	struct bt_conn *conn;
+	int err;
+
+	LOG_DBG("MCC Send Command");
+
+	conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, &cp->address);
+	if (!conn) {
+		LOG_ERR("Unknown connection");
+		return BTP_STATUS_FAILED;
+	}
+
+	mcp_cmd.opcode = cp->opcode;
+	mcp_cmd.use_param = cp->use_param;
+	mcp_cmd.param = sys_le32_to_cpu(cp->param);
+
+	err = bt_mcc_send_cmd(conn, &mcp_cmd);
+	if (err) {
+		return BTP_STATUS_FAILED;
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t mcp_cmd_search(const void *cmd, uint16_t cmd_len, void *rsp, uint16_t *rsp_len)
+{
+	const struct btp_mcp_search_cmd *cp = cmd;
+	struct mpl_search search_items;
+	struct mpl_sci scp_cmd;
+	struct bt_conn *conn;
+	int err;
+
+	LOG_DBG("MCC Send Search Control Point Command");
+
+	if (cmd_len < sizeof(*cp) || cmd_len != sizeof(*cp) + cp->param_len) {
+		return BTP_STATUS_FAILED;
+	}
+
+	conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, &cp->address);
+	if (!conn) {
+		LOG_ERR("Unknown connection");
+		return BTP_STATUS_FAILED;
+	}
+
+	search_items.len = 0;
+	scp_cmd.type = cp->type;
+
+	if (scp_cmd.type == BT_MCS_SEARCH_TYPE_ONLY_TRACKS ||
+	    scp_cmd.type == BT_MCS_SEARCH_TYPE_ONLY_GROUPS) {
+		scp_cmd.len = sizeof(scp_cmd.type);
+		memcpy(&search_items.search[search_items.len], &scp_cmd.len, sizeof(scp_cmd.len));
+		search_items.len += sizeof(scp_cmd.len);
+
+		memcpy(&search_items.search[search_items.len], &scp_cmd.type,
+		       sizeof(scp_cmd.type));
+		search_items.len += sizeof(scp_cmd.type);
+	} else {
+		if (cp->param_len >= (SEARCH_LEN_MAX - 1)) {
+			return BTP_STATUS_FAILED;
+		}
+
+		strcpy(scp_cmd.param, cp->param);
+		scp_cmd.len = sizeof(scp_cmd.type) + strlen(scp_cmd.param);
+
+		memcpy(&search_items.search[search_items.len], &scp_cmd.len, sizeof(scp_cmd.len));
+		search_items.len += sizeof(scp_cmd.len);
+
+		memcpy(&search_items.search[search_items.len], &scp_cmd.type,
+		       sizeof(scp_cmd.type));
+		search_items.len += sizeof(scp_cmd.type);
+
+		memcpy(&search_items.search[search_items.len], &scp_cmd.param,
+		       sizeof(scp_cmd.param));
+		search_items.len += strlen(scp_cmd.param);
+
+		if (search_items.len > ARRAY_SIZE(search_items.search)) {
+			return BTP_STATUS_FAILED;
+		}
```

Comment:
You need to check this before you access `search_items.search[search_items.len]`, and also in the above `if clause`